pagopa/io-app
# fix(Firma con IO): [SFEQS-2124] Wrong margins in FCI flow

## Short description
This PR fixes margins in FCI flow screens. 

| Before | After |
|--------|--------|
| <img src="https://github.com/pagopa/io-app/assets/12371664/8a3c5192-8ceb-4ce4-bfc0-24ce6a8a1691" height="350"> | <img src="https://github.com/pagopa/io-app/assets/12371664/a9210c57-b2d7-4b01-8dfe-9cce669d83c8" height="350"> |
| <img src="https://github.com/pagopa/io-app/assets/12371664/95d0d142-abfc-4b46-bf9d-3229117b2093" height="350"> | <img src="https://github.com/pagopa/io-app/assets/12371664/4c9cf5ee-c8ec-493b-889b-d84082bca837" height="350"> |
| <img src="https://github.com/pagopa/io-app/assets/12371664/f9b8e474-d858-48a5-b69e-ba991779041a" height="350"> | <img src="https://github.com/pagopa/io-app/assets/12371664/807f61fa-6769-4243-bb41-edad27f988a8" height="350"> |

## List of changes proposed in this pull request
- Removes `SafeAreaView` in screens where there's a `useHeaderSecondLevel` hook and a `FooterWithButtons` component because they already consume safe area insets. This prevents excessive margins. See first comparison.
- Keeps `SafeAreaView` with every edge in screens where there's no `useHeaderSecondLevel` hook nor a `FooterWithButtons` component. Keeps `SafeAreaView` with `top` edge where there's no `useHeaderSecondLevel` and/or `bottom` edge where there's no `FooterWithButtons` component. See second comparison;
- Replaces `FooterWithButtons` in `useFciAbortSignatureFlow` with `BlockButtons` because the bottom sheet already consumes insets for the bottom padding, thus preventing double insets. See third comparison.

## How to test
Test every single screen on both iOS and Android including success, error and loading views on the FCI flow and check for any regression.
Please note that there's an issue with the abort bottom sheet on Android which is not showing a bottom padding. This is due to the bottom sheet logic which doesn't add a default value if the the inset is 0 (as already done in `FooterWithButtons`). That's out of scope of this PR.
